src/hb_gateway_client.erl

```diff
@@ -424,11 +425,28 @@ l1_transaction_test() ->
 %% @doc Test l2 message from graphql
 l2_dataitem_test() ->
     _Node = hb_http_server:start_node(#{}),
-    {ok, Res} = read(<<"oyo3_hCczcU7uYhfByFZ3h0ELfeMMzNacT-KpRoJK6g">>, #{}),
+    ID = <<"oyo3_hCczcU7uYhfByFZ3h0ELfeMMzNacT-KpRoJK6g">>,
```

Comment:
Why unwrap this?

Reply:
I added a test to validate commitment type in `l2_dataitem_ed25519_test`. Since I didn't find any test with the RSA key, I've also added this check on one of the other tests. Since I need the ID to access to the value inside the message, I decided to extract the binary to a variable to only write the ID value once.

Reply:
`{ok, Res} = read(ID = <<"oyo3_hCczcU7uYhfByFZ3h0ELfeMMzNacT-KpRoJK6g">>, #{}),` is the general way we do things like this, if the lines don't become unwieldy.